Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,141 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+#include <boost/algorithm/string.hpp>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "MolSupplier.h"
+#include "MultithreadedSmilesMolSupplier.h"
+#include "MultithreadedSDMolSupplier.h"
+
+namespace RDKit {
+namespace GeneralMolSupplier {
+struct SupplierOptions {
+  bool takeOwnership = true;
+  bool sanitize = true;
+  bool removeHs = true;
+  bool strictParsing = true;
+
+  std::string delimiter = "\t";
+  int smilesColumn = 0;
+  int nameColumn = 1;
+  bool titleLine = true;
+
+  std::string nameRecord = "";
+  int confId2D = -1;
+  int confId3D = 0;
+
+	unsigned int numWriterThreads = 0;
+};
+//! current supported file formats
+std::vector<std::string> supportedFileFormats{"sdf", "mae", "maegz", "smi",
+                                              "csv", "txt", "tsv",   "tdt"};
+//! current supported compression formats
+std::vector<std::string> supportedCompressionFormats{"gz"};
+
+//! given file path determines the file and compression format
+void determineFormat(const std::string path, std::string& fileFormat,
```

Comment:
I think it would be more robust to make this return enum class objects. Relying on string comparison leaves you vulnerable to coding errors via typos that would only be caught at runtime. Enum class comparisons would be caught at compile time! Also, `getSupplier()` could then use a switch statement.

    std::pair<FileFormat, CompressionFormat> guessFormat(const& std::string path);

Use like (C++17 only):

    auto [format, compression] = guessFormat(path);

Reply:
Worth considering for a future enhancement.